protocol/communities/manager.go

```diff
@@ -1478,6 +1486,64 @@ func (m *Manager) HandleCommunityEventsMessageRejected(signer *ecdsa.PublicKey,
 	return reapplyEventsMessage, nil
 }
 
+func (m *Manager) HandleCommunityPrivilegedUserSyncMessage(signer *ecdsa.PublicKey, message *protobuf.CommunityPrivilegedUserSyncMessage) error {
+	if signer == nil {
+		return errors.New("signer can't be nil")
+	}
+
+	community, err := m.persistence.GetByID(&m.identity.PublicKey, message.CommunityId)
+	if err != nil {
+		return err
+	}
+
+	if community == nil {
+		return ErrOrgNotFound
+	}
+
+	if !community.IsPrivilegedMember(&m.identity.PublicKey) {
+		return errors.New("user has no permissions to process privileged sync message")
+	}
+
+	if !(common.IsPubKeyEqual(community.PublicKey(), signer) || community.IsPrivilegedMember(signer)) {
+		return errors.New("user has no permissions to send privileged sync message")
+	}
+
+	err = validateCommunityPrivilegedUserSyncMessage(message)
+	if err != nil {
+		return err
+	}
+
+	switch message.Type {
+	case protobuf.CommunityPrivilegedUserSyncMessage_CONTROL_NODE_ACCEPT_REQUEST_TO_JOIN:
+		fallthrough
+	case protobuf.CommunityPrivilegedUserSyncMessage_CONTROL_NODE_REJECT_REQUEST_TO_JOIN:
+		var state RequestToJoinState
+		if message.Type == protobuf.CommunityPrivilegedUserSyncMessage_CONTROL_NODE_ACCEPT_REQUEST_TO_JOIN {
+			state = RequestToJoinStateAccepted
+		} else {
+			state = RequestToJoinStateDeclined
```

Comment:
Ok, so if I understand correctly, the new message type for privileged users is **only** sent by the control node?

I don't think that's exactly what the issue was asking. I think it was a type of message that privileged users can send each other, like a small group chat that normal members are not part of. 

Maybe @0x-r4bbit can confirm?

Reply:
no, it can be sent by any privileged users, the sender decides who will receive this message :) 
In this case, it makes sense to verify that this 2 types of msg (accepted/rejected) comes from the control node